doc/releases/changelog-dev.md

```diff
@@ -4,34 +4,125 @@
 
 <h3>New features since last release</h3>
 
-* Devices have a new capability flag `capabilities()["supports_broadcasting"]`
-  and are now able to handle broadcasting of tapes.  In addition, the tape transform
-  `broadcast_expand` was added, which allows a tape that uses broadcasting 
-  (`tape.batch_size!=None`) to be expanded into multiple tapes without a broadcast dimension.
+* Parameter broadcasting within operations and tapes was introduced.
+
+  [(#2575)](https://github.com/PennyLaneAI/pennylane/pull/2575)
   [(#2590)](https://github.com/PennyLaneAI/pennylane/pull/2590)
+  [(#2609)](https://github.com/PennyLaneAI/pennylane/pull/2609)
+
+  Parameter broadcasting refers to passing parameters with a (single) leading additional
+  dimension (compared to the expected parameter shape) to `Operator`s.
+  Introducing this concept includes multiple changes:
+
+  1. New class attributes
+    - `Operator.ndim_params` contains the expected number of dimensions for each parameter
+      of an operator.
+    - `Operator.batch_size` contains the size of an additional parameter-broadcasting axis,
+      if present.
+    - `QuantumTape.batch_size` contains the `batch_size` of its operations (see logic below).
```

Comment:
```suggestion
    - `Operator.batch_size` returns the size of an additional parameter-broadcasting axis,
      if present.
    - `QuantumTape.batch_size` returns the `batch_size` of its operations (see logic below).
```

Reply:
to try and make clear that these don't need to be defined, whereas `ndim_params` do